### PR TITLE
Implement Better Terminal Behavior

### LINKED ISSUE #227: Terminal output scroll
Current experience:
When opening the terminal output during the installation, the window always shows the top/beginning of the output. Manual scrolling is required to see the latest progress of the installation.

Expected experience:
When selecting the terminal button, the output should automatically scroll to the bottom to show the most recent output. Whenever this screen is scrolled all the way down, it should continue to scroll when new input is added. If the user scrolls up, automatic scrolling should stop.

### PR DESCRIPTION
**Pull request by mmstick**
Closes #227

- Makes a change to how the log buffer is handled. Concatenating text to the buffer directly causes the text view to reset, losing the scrolled position. Instead, using the `insert` method on the buffer allows the existing text contents to be retained, and new contents inserted at the end.
- The terminal button will now automatically scroll to the bottom of the view when the inner text view is updated, and the view was previously at the bottom.

### CODE REVIEW

File: src/Views/ProgressView.vala [hunk 1]
```diff
@@ -74,14 +75,31 @@ public class ProgressView : AbstractInstallerView {
...
+                scroll_to_bottom ();
+            }
+        });
+
         show_all ();
     }
 
+    private void scroll_to_bottom () {
+        var adj = terminal_output.vadjustment;
+        adj.value = adj.upper;
```
Comment (jackpot51) [inline src/Views/ProgressView.vala]: Should this be:
```
adj.value = adj.upper - adj.page_size;
```

**Comment by jackpot51** [comment]
This is not quite working for me, and I am not sure why. I opened the terminal immediately after getting to the progress view, and it scrolled for about two lines before no longer scrolling.

File: src/Views/ProgressView.vala (hunk 1 above)
Comment (jackpot51) [reply src/Views/ProgressView.vala]: I don't think the change I proposed works to fix the issues I have seen either

**Comment by mmstick** [comment]
I may need to account for the height of the last line(s) added, in case there's more than one.

**Comment by jackpot51** [comment]
Yep, that sounds right

**Comment by mmstick** [comment]
After much mangling with values, I think I've got it. Need to test a bit more, though.

**Comment by jackpot51** [comment]
Sounds good @mmstick. I will take a look when it is committed

**Comment by jackpot51** [comment]
That looks better @mmstick

**Comment by jackpot51** [comment]
There is a point early in the install process where distinst outputs more than one page of content, and it does track the bottom of the output

**Comment by jackpot51** [comment]
If you add a GLib.Thread.usleep(1000000) to the start_installation function, you will have time to open the log view before any output

**Comment by mmstick** [comment]
Going to try a different approach.

**Comment by mmstick** [comment]
It took countless hours, and multiple attempts with different approaches, but the new solution should work better. Modified distinst locally to generate all sorts of random log data of varying numbers of newlines, at random intervals (from bursts multiple pages long to once a second).

The only thing I haven't figured out is how to prevent the ScrolledWindow from rubber banding back when the bottom end is overshot. Using the scrollbar to scroll to the end doesn't cause that behavior. Couldn't find any documentation from GTK or elsewhere on the matter.

**Pull request merged**